src/components/AddItem.jsx

```diff
@@ -1,27 +1,58 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { collection, addDoc } from 'firebase/firestore';
 import { db } from '../lib/firebase';
 import { getUser } from '../storage-utils/storage-utils';
+import { onSnapshot } from 'firebase/firestore';
 
 export default function AddItem() {
   const [userToken] = useState(getUser());
   const [itemName, setItemName] = useState('');
   const [frequency, setFrequency] = useState(7);
+  const [message, setMessage] = useState('');
+  const [docs, setDocs] = useState([]);
+
+  useEffect(() => {
+    const unsubscribe = onSnapshot(collection(db, userToken), (snapshot) => {
+      let snapshotDocs = [];
+      snapshot.forEach((doc) => snapshotDocs.push(doc));
+      setDocs(snapshotDocs);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [userToken]);
 
   const submitHandler = async (e) => {
     e.preventDefault();
 
+    const cleanList = docs.map((listItem) =>
+      listItem
+        .data()
+        .item.replace(/[^a-zA-Z\s]/g, '')
+        .replace(/\s+/g, '')
```

Comment:
@nikkigraybeal I just updated lines 32 and 38 to replace them with an empty string and I think that fixes the issue? Can you test on your end as well?

Reply:
yes! I tested with that fix and it seems to work no matter what spaces and special chars I include 🎉